rulesets/java/jpinpoint-rules.xml

```diff
@@ -326,12 +326,13 @@ class Good {
 (: only if the number of parameters match: not an overloaded method with difference in numbers :)
 [ArgumentList/@Size = ancestor::MethodDeclaration/FormalParameters/@Size]
 (: not if any parameter is used in the call in another position :)
-[not(ArgumentList/VariableAccess[((position() != index-of(ancestor::MethodDeclaration/FormalParameters/FormalParameter/VariableId/@Name, @Name)))])]
+[not((ArgumentList/*[self::VariableAccess and ((position() != index-of(ancestor::MethodDeclaration/FormalParameters/FormalParameter/VariableId/@Name, @Image)))]))]
```

Comment:
@Image use is deprecated in PMD 7? [See](https://pmd.github.io/pmd/pmd_userdocs_migrating_to_pmd7.html#general-ast-changes-to-avoid-image): "For PMD 7, most languages have been adapted. And when writing XPath rules, you need to replace @Image with whatever is appropriate now (e.g. @Name)."

Reply:
Right, seems to be possible to use concat(Element/@Name, '') to make sure there is at an empty value, otherwise I get an xpath error